### PR TITLE
Replace bundled styles with mapbox: style URLs

### LINKED ISSUE #2239: unbundle default Mapbox styles from SDKs
Load these via the Mapbox Styles API instead. Bundling doesn't win us anything, because network access is still needed for sources, sprite, fonts, and tiles. Unbundling allows us to remove the styles submodule from this repository (and not replace it with an npm dependency).

### PR DESCRIPTION
**Pull request by ljbade**
Fixes #2239

### CODE REVIEW

**Comment by 1ec5** [comment]
We also need to deprecate or remove the bundledStyleURLs property from MGLMapView, since that property always returns an empty array. We can deprecate it with a compile-time message instructing developers to look at some webpage for a list of always-available styles, but we need to have that webpage ready.

**Comment by ljbade** [comment]
@1ec5 can we make it return a list of `mapbox://` URLs?

**Comment by 1ec5** [comment]
That’s not a bad idea. We’d just base it off default_styles.cpp, which would enable us to nix the parallel list we’ve compiled in MBXViewController. We’d still need to rename the method, though: the styles aren’t “bundled”, just universally available, regardless of access token.

**Comment by ljbade** [comment]
@1ec5 I was thinking more from source compatibility perspective but next release is breaking anyway right?

**Comment by 1ec5** [comment]
You’re right, we should mark `-bundledStyleURLs` as deprecated with a message to migrate to whatever we call it. `-universalStyleURLs`?

**Comment by ljbade** [comment]
`-mapboxStyleURLs`?

**Comment by 1ec5** [comment]
I’ve completed the iOS side of this change. There’s a new class MGLStyle that – for now – does nothing but vend style URLs in a manner similar to UIColor. It would make for a great entry point an iOS runtime styling API as part of #837. Unlike Style.java on the Android side, MGLStyle gets its values directly from a refactored default_styles.hpp and uses static assertions to stay in sync. So that takes care of #1462. Objective-C++, oh happy day!

The upshot for other platforms is that there are now individual constants for each “default style” (that is, a style that’s available regardless of access token). There’s still a constant C array of them in case you need to iterate over them.

@jfirebaugh, does my C++ look reasonable? @incanus, can you forgive my macro shenanigans?

File: include/mbgl/util/default_styles.hpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+#include <vector>
+#include <string>
+
+namespace mbgl {
+namespace util {
+namespace default_styles {
+
+struct DefaultStyle {
+    const char *url;
```
Comment (jfirebaugh) [inline include/mbgl/util/default_styles.hpp]: Our style for C++ is to space `*` and `&` with the rest of the type, so `const char* url`.

**Comment by jfirebaugh** [comment]
LGTM other than the one nit.

File: include/mbgl/util/default_styles.hpp (hunk 1 above)
Comment (1ec5) [inline include/mbgl/util/default_styles.hpp]: Old habits die hard: e13535eda83116ced526c6785e9596e25bdfaac2.

**Comment by ljbade** [comment]
Excellent, when your happy @1ec5 I think we can squash, get Travis green, final set of eyes, then merge.

**Comment by 1ec5** [comment]
Running into #2022.

**Comment by ljbade** [comment]
@1ec5 is that the right ticket number?

**Comment by 1ec5** [comment]
Yes, that was precisely what I and Travis were both hitting, but now it’s green again. Go figure.

**Comment by 1ec5** [comment]
I’ve come back around to the idea of deprecating and eventually removing `-bundledStyleURLs` from the iOS SDK (rather than just renaming it). I don’t see a real use case for host applications to iterate over these style URLs without any names that would be suitable for display in a UI. With the introduction of `-[MGLStyle streetsStyleURL]` et al., developers no longer need to rely on `-bundledStyleURLs` to [discover available styles](https://www.mapbox.com/ios-sdk/examples/custom-style/).

**Pull request merged**